src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/custom.py

```diff
@@ -0,0 +1,23 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def natgateway_client_factory(cli_ctx, **kwargs):
+    from azure.cli.core.profiles import ResourceType
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK, **kwargs)
+
+def create_nat_gateway(cmd, nat_gateway_name, resource_group_name, location, public_ip_address, public_ip_prefix=None, idle_timeout=None):
+
+    client = natgateway_client_factory(cmd.cli_ctx).nat_gateways
+    from azure.mgmt.network.models import NatGateway, NatGatewaySku, PublicIPAddress, PublicIPPrefix
```

Comment:
Network is a multi-API SDK. This needs to use the import format used in the network module:

```Python
NatGateway, NatGatewaySku, … = cmd.get_models('NatGateway', 'NatGatewaySku', …)
```